channel/handlers_client.go

```diff
@@ -1325,6 +1327,214 @@ func (server *Server) playerUseScroll(conn mnet.Client, reader mpacket.Reader) {
 	}
 }
 
+func (server *Server) playerUseCash(conn mnet.Client, reader mpacket.Reader) {
+	plr, err := server.players.GetFromConn(conn)
+	if err != nil {
+		return
+	}
+
+	slot := reader.ReadInt16()
+	itemID := reader.ReadInt32()
+
+	used := false
+
+	switch itemID {
+	case constant.ItemSafetyCharm:
+		plr.hasSafetyCharm = true
+		used = true
+
+	case constant.ItemAPReset:
+		statUp := reader.ReadInt32()
+		statDown := reader.ReadInt32()
+
+		if (statUp == constant.StrID || statUp == constant.DexID || statUp == constant.IntID || statUp == constant.LukID) &&
+			(statDown == constant.StrID || statDown == constant.DexID || statDown == constant.IntID || statDown == constant.LukID) &&
+			statUp != statDown {
+			canReset := false
+			switch statDown {
+			case constant.StrID:
+				canReset = plr.str > 4
+			case constant.DexID:
+				canReset = plr.dex > 4
+			case constant.IntID:
+				canReset = plr.intt > 4
+			case constant.LukID:
+				canReset = plr.luk > 4
+			}
+
+			if canReset {
+				switch statDown {
+				case constant.StrID:
+					plr.setStr(plr.str - 1)
+				case constant.DexID:
+					plr.setDex(plr.dex - 1)
+				case constant.IntID:
+					plr.setInt(plr.intt - 1)
+				case constant.LukID:
+					plr.setLuk(plr.luk - 1)
+				}
+
+				switch statUp {
+				case constant.StrID:
+					plr.setStr(plr.str + 1)
+				case constant.DexID:
+					plr.setDex(plr.dex + 1)
+				case constant.IntID:
+					plr.setInt(plr.intt + 1)
+				case constant.LukID:
+					plr.setLuk(plr.luk + 1)
+				}
+
+				used = true
+			}
+		}
+
+	case constant.ItemSPResetFirstJob, constant.ItemSPResetSecondJob, constant.ItemSPResetThirdJob:
+		skillUp := reader.ReadInt32()
+		skillDown := reader.ReadInt32()
+
+		skillUpData, okUp := plr.skills[skillUp]
+		skillDownData, okDown := plr.skills[skillDown]
+
+		if okUp && okDown && skillDownData.Level > 0 {
+			skillDownData.Level--
+			if skillDownData.Level == 0 {
+				delete(plr.skills, skillDown)
+			} else {
+				plr.updateSkill(skillDownData)
+			}
+
+			skillUpData.Level++
+			plr.updateSkill(skillUpData)
+
+			used = true
+		}
+
+	case constant.ItemVIPTeleportRock, constant.ItemRegTeleportRock:
+		mode := reader.ReadByte()
+
+		if mode == 0x01 {
```

Comment:
Should probably make it a constant like the other magic numbers. At some point it needs to be done to the older parts of the code base 🥲